File: 1359. Count All Valid Pickup and Delivery Options/count-all-valid-pickup-and-delivery-options.ts
// 1359. Count All Valid Pickup and Delivery Options (3/6/2022)
// Runtime: 60 ms (0.00%) Memory: 44.80 MB (0.00%) 

function countOrders(n: number): number {
    let ans = 1;
    const mod = 1000000007;
    
    for (let i = 1; i <= n; i++) {
        ans *= 2 * i - 1;
        ans %= mod;
    }
    
    for (let i = 1; i <= n; i++) {
        ans *= i;
        ans %= mod;
    }
    
    return ans;
};

File: 1359. Count All Valid Pickup and Delivery Options/count-all-valid-pickup-and-delivery-options.rb
# 1359. Count All Valid Pickup and Delivery Options (3/7/2022)
# Runtime: 48 ms (0.00%) Memory: 210.95 MB (0.00%) 

# @param {Integer} n
# @return {Integer}
def count_orders(n)
    ans = 1
    mod = 1000000007
    
    for i in 2..n
        ans *= i
        ans %= mod
    end
    
    for i in 2..n
        ans *= i * 2 - 1
        ans %= mod
    end
    
    return ans
end
